commit 82aba1ea2b11362216d077f6439530cdacb242dd (refs/changes/55/2655/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-09-26T12:09:47-07:00 (2 years ago)
    
    PUBAPI-1445 improve CreateMachine affinity error message if a "==" affinity matches no VMs

diff --git a/lib/triton-affinity.js b/lib/triton-affinity.js
index e41fd4e..43478e9 100644
--- a/lib/triton-affinity.js
+++ b/lib/triton-affinity.js
@@ -956,6 +956,12 @@ function localityFromRulesInfo(opts) {
     });
     var nearServerUuids = null; // Servers that satisfy all near rules.
     nearRules.forEach(function (nearRule) {
+        if (nearRule.vms.length === 0) {
+            throw new VError('cannot satisfy affinity rule "%s", '
+                + 'it does not match any instances',
+                exprFromRule(nearRule));
+        }
+
         // Eliminate "far" servers from candidacy.
         nearRule.remainingVms = nearRule.vms.filter(function (vm) {
             // Provisioning VMs might not have a server_uuid.
